### ISSUE TITLE
the image is not shown

### ISSUE THREAD

**Issue by ortiga88** [grafana/grafana-image-renderer#166]
I have a docker-compose:

```
 renderer:
    image: grafana/grafana-image-renderer:2.0.0
    container_name: renderer_new
    restart: unless-stopped
    environment:
      - BROWSER_TZ=Europe/Madrid
      - RENDERING_MODE=default
    user: "0"
    logging:
      driver: "json-file"
      options:
        max-size: 50m
    environment:
      ENABLE_METRICS: "true"
    networks:
      - grafana

```

in grafana:

```
 - GF_RENDERING_SERVER_URL=http://renderer:8..../render
 - GF_RENDERING_CALLBACK_URL=http://grafana:4...../
 - GF_LOG_FILTERS=rendering:debug

```

error:

```

{"msg":"Failed to load resource: the server responded with a status of 401 (Unauthorized)","url":"http://grafana:4.../d-solo/qk2Pk8WMk/jaeger?orgId=1&panelId=10&render=1","level":"error","message":"Browser console error"}
{"url":"/render?deviceScaleFactor=1.000000&domain=grafana&encoding=&height=500&renderKey=cqJWcR684PsNY1tI5OI6NLCEUu7dn4sX&timeout=2&timezone=&url=http%3A%2F%2Fgrafana%3A3000%2Fd-solo%2Fqk2Pk8WMk%2Fjaeger%3ForgId%3D1%26panelId%3D10%26render%3D1&width=1000","stack":"TimeoutError: waiting for function failed: timeout 2000ms exceeded\n    at new WaitTask (/usr/src/app/node_modules/puppeteer/lib/DOMWorld.js:549:28)\n    at DOMWorld.waitForFunction (/usr/src/app/node_modules/puppeteer/lib/DOMWorld.js:454:12)\n    at Frame.waitForFunction (/usr/src/app/node_modules/puppeteer/lib/FrameManager.js:657:28)\n    at Page.waitForFunction (/usr/src/app/node_modules/puppeteer/lib/Page.js:1144:29)\n    at Browser.<anonymous> (/usr/src/app/build/browser/browser.js:174:24)\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/src/app/build/browser/browser.js:5:58)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error","message":"Request failed"}
```

**Comment by todfox** [comment]
Also having this issue.

**Comment by bibarrav** [comment]
Same here!

**Comment by bibarrav** [comment]
Check #151 I put a lot of info, and finally I resolve a similar problem on my setup.  I hope it can work for you!

**Comment by AgnesToulet** [comment]
As mentioned here https://github.com/grafana/grafana-image-renderer/issues/151#issuecomment-910462759:

> Often, when having performance issues, the 401 error will be returned because the Grafana request will timeout and delete the renderKey before the rendering request reaches the point of navigating to the Grafana page.
> 
> A fix to this issue #186 would fix this misleading error.

Also, it appears like the timeout of the rendering request is set to 2 seconds here. It is too small to render a panel (as it requires to launch a Chromium browser / page, navigate to the page, wait for the data source query to end and take a screenshot). You should increase it. 

To update the timeout of a rendering request, it depends on the request type:
- For alerting, it's this config setting (divided by 2): https://grafana.com/docs/grafana/latest/administration/configuration/#notification_timeout_seconds
- For reporting, it's this one: https://grafana.com/docs/grafana/latest/enterprise/reporting/#rendering-configuration
- For panel rendering, it's in the rendering request itself, for example: https://<grafana>/render/d-solo/XJHWWsLMk/grafana-cloud-billing-usage?orgId=[…]420798553&panelId=5&width=1000&height=500&tz=Europe%2FParis&timeout=30 (default is set to 30 that is more than enough if you don't have too long queries).
